s3/pkg/meta/db/drivers/tidb/object.go

```diff
@@ -1,20 +1,23 @@
 package tidbclient
 
 import (
+	"context"
 	"database/sql"
-	. "github.com/opensds/multi-cloud/s3/pkg/meta/types"
+	"encoding/hex"
+	"encoding/json"
 	"math"
 	"strconv"
-	"encoding/hex"
+	"time"
+
+	. "github.com/opensds/multi-cloud/s3/pkg/meta/types"
+	pb "github.com/opensds/multi-cloud/s3/proto"
+	log "github.com/sirupsen/logrus"
 	"github.com/xxtea/xxtea-go/xxtea"
-	. "github.com/opensds/multi-cloud/s3/error"
-	"context"
 )
 
 func (t *TidbClient) GetObject(ctx context.Context, bucketName, objectName, version string) (object *Object, err error) {
-	var ibucketname, iname, customattributes, acl, lastModifiedTime string
+	var sqltext, ibucketname, iname, customattributes, acl, lastModified string
 	var iversion uint64
-	var sqltext string
 	var row *sql.Row
 	if version == "" {
 		sqltext = "select * from objects where bucketname=? and name=? order by bucketname,name,version limit 1;"
```

Comment:
is this ordering correct ? or do we need to show the latest versions of all objects first, like below, do we show as 1 or 2 ? (b1 is bucket'n', o1 is object'n' and v1 is version'n' for illustration)
b1-o1-v3
b1-o1-v2
b1-o1-v1
b1-o2-v3

OR

b1-o1-v3
b1-o2-v3
b1-o1-v2
b1-o1-v2
b1-o1-v1
b1-o2-v1

get what i mean ? if one object has many versions, it should NOT push other objects to the bottom

Reply:
Yes, you are right, if versioning is enabled, then we need to show the latest versions of all objects first. But currently, there is no versioning, and we have issue for versioning, see [#713](https://github.com/opensds/multi-cloud/issues/713) please.